app/main.py

```diff
@@ -1,3 +1,50 @@
 class Distance:
-    # Write your code here
-    pass
+    def __init__(self, km: int):
+        self.km = km
+
+    def __repr__(self):
+        return f"Distance(km={self.km})"
+
+    def __str__(self):
+        return f"Distance: {self.km} kilometers"
+
+    def __add__(self, other):
+        if isinstance(other, (float, int)):
+            return Distance(self.km + other)
+        return Distance(self.km + other.km)
+
+    def __iadd__(self, other):
+        if isinstance(other, (float, int)):
+            self.km += other
+        else:
+            self.km += other.km
+        return self
```

Comment:
same here

Reply:
When I changed if-statement in __iadd__ and __add__ , it returned me "TypeError: unsupported operand type(s) for +: 'int' and 'Distance'", so I don't understand, how I can fix it.

Reply:
Try to write like this:

```
       if isinstance(other, Distance):
            self.km += other.km
            return self
        elif isinstance(other, (int, float)):
            self.km += other
            return self
```